### PR TITLE
Feature/auth refresh and jwt support

### PR DESCRIPTION
**Pull request by chrisdp**

### CODE REVIEW

File: source/components/AblyTask/utilities.brs [hunk 1]
```diff
@@ -86,9 +86,130 @@ end function
...
+    "url": requestUrl
+  }
+end function
+
+function createNewUrlTransfer(messagePort as Object) as Object
+  transferObject = createObject("roUrlTransfer")
+  transferObject.setPort(messagePort)
+  transferObject.enableEncodings(true)
+  transferObject.retainBodyOnError(true)
+  ' transferObject.setMinimumTransferRate(2147483647, m.timeout)
```
Comment (QuintinWillison) [inline source/components/AblyTask/utilities.brs:210]: Looks like this line is commented out but without any surrounding meta-commentary to explain why.

Could you either add some commentary or remove it.

File: README.md [hunk 2]
```diff
@@ -43,6 +43,9 @@ sub init()
   ' Create the AblyTask
   m.ablyTask = createObject("roSGNode", "AblyTask")
 
+  ' Assign the API to the task used for authentication
+  m.ablyTask.key = "xVLyHw.XuwW-w:yOPxtXWlsZn10nzy"
```
Comment (QuintinWillison) [inline README.md:47]: Where did this key come from? It _looks_ 'real' which is a bad idea. Can you replace with some kind of clearly redacted placeholder?

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
I'm not going to find time today to spin this up on my BrightSign unit so I'll trust you that it works! I will aim to give it a spin soon. In the meantime, a couple of observations.

File: source/components/AblyTask/utilities.brs (hunk 1 above)
Comment (chrisdp) [reply source/components/AblyTask/utilities.brs:210]: I think I can remove it. Was left in from the POC days and we don't really need a hard timeout I don't think. If we need it later I can add it back.

File: README.md (hunk 2 above)
Comment (chrisdp) [reply README.md:47]: Taken from a example page like this one: https://www.ably.io/documentation/core-features/authentication#basic-authentication

Mind you I don't remember witch one I grabbed it from.

File: source/components/AblyTask/utilities.brs (hunk 1 above)
Comment (chrisdp) [reply source/components/AblyTask/utilities.brs]: 882a3d9

**Review by QuintinWillison — APPROVED** [review]

**Pull request merged**